From: Kevin Barnett <kevin.barnett@microsemi.com>
Date: Wed, 27 Sep 2017 16:30:05 -0500
Subject: scsi: smartpqi: cleanup raid map warning message
Patch-mainline: v4.15-rc1
Git-commit: 38a7338ab55a1e1e2e86113eb5694350c3a2c865
References: FATE#323891,bsc#1049514

Fix a small cosmetic bug in a very rarely encountered error message that
can occur when a LD has a corrupted raid map.

Reviewed-by: Scott Benesh <scott.benesh@microsemi.com>
Reviewed-by: Tomas Henzl <thenzl@redhat.com>
Signed-off-by: Kevin Barnett <kevin.barnett@microsemi.com>
Signed-off-by: Don Brace <don.brace@microsemi.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/scsi/smartpqi/smartpqi_init.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/scsi/smartpqi/smartpqi_init.c
+++ b/drivers/scsi/smartpqi/smartpqi_init.c
@@ -1078,9 +1078,9 @@ static int pqi_validate_raid_map(struct
 
 bad_raid_map:
 	dev_warn(&ctrl_info->pci_dev->dev,
-		"scsi %d:%d:%d:%d %s\n",
-		ctrl_info->scsi_host->host_no,
-		device->bus, device->target, device->lun, err_msg);
+		"logical device %08x%08x %s\n",
+		*((u32 *)&device->scsi3addr),
+		*((u32 *)&device->scsi3addr[4]), err_msg);
 
 	return -EINVAL;
 }
